sklearn/feature_extraction/text.py

```diff
@@ -852,7 +852,7 @@ def _sort_features(self, X, vocabulary):
         Returns a reordered matrix and modifies the vocabulary in place
```

Comment:
Unrelated, but: This seems to modify `X` in place too.

Reply:
That's intentional, I think, to reduce memory usage, why would it be an issue?

Reply:
It's an issue only because the documentation doesn't tell me that's going to happen.

Reply:
> This seems to reorder X in place too.

Agreed, we should add a note about it but in a separate PR.